pkg/workloads/k8s.go

```diff
@@ -0,0 +1,208 @@
+		Flist:         d.FList,
+		FlistChecksum: flistCheckSum,
+		ComputedIP:    computedIP,
+		ComputedIP6:   computedIP6,
+		YggIP:         result.YggIP,
+		IP:            d.Network.Interfaces[0].IP.String(),
+		CPU:           int(d.ComputeCapacity.CPU),
+		Memory:        int(d.ComputeCapacity.Memory / gridtypes.Megabyte),
+	}
+	return k, nil
+}
+
+func (k *K8sNodeData) Dictify() map[string]interface{} {
+	res := make(map[string]interface{})
+	res["name"] = k.Name
+	res["node"] = int(k.Node)
+	res["disk_size"] = k.DiskSize
+	res["publicip"] = k.PublicIP
+	res["publicip6"] = k.PublicIP6
+	res["planetary"] = k.Planetary
+	res["flist"] = k.Flist
+	res["flist_checksum"] = k.FlistChecksum
+	res["computedip"] = k.ComputedIP
+	res["computedip6"] = k.ComputedIP6
+	res["ygg_ip"] = k.YggIP
+	res["ip"] = k.IP
+	res["cpu"] = k.CPU
+	res["memory"] = k.Memory
+	return res
+}
+
+func (k *K8sNodeData) GenerateK8sWorkload(cluster *K8sCluster, masterIP string) []gridtypes.Workload {
+	diskName := fmt.Sprintf("%sdisk", k.Name)
+	K8sWorkloads := make([]gridtypes.Workload, 0)
+	diskWorkload := gridtypes.Workload{
+		Name:        gridtypes.Name(diskName),
+		Version:     0,
+		Type:        zos.ZMountType,
+		Description: "",
+		Data: gridtypes.MustMarshal(zos.ZMount{
+			Size: gridtypes.Unit(k.DiskSize) * gridtypes.Gigabyte,
+		}),
+	}
+	K8sWorkloads = append(K8sWorkloads, diskWorkload)
+	publicIPName := ""
+	if k.PublicIP || k.PublicIP6 {
+		publicIPName = fmt.Sprintf("%sip", k.Name)
+		K8sWorkloads = append(K8sWorkloads, ConstructPublicIPWorkload(publicIPName, k.PublicIP, k.PublicIP6))
+	}
+	envVars := map[string]string{
+		"SSH_KEY":           cluster.SSHKey,
+		"K3S_TOKEN":         cluster.Token,
+		"K3S_DATA_DIR":      "/mydisk",
+		"K3S_FLANNEL_IFACE": "eth0",
+		"K3S_NODE_NAME":     k.Name,
+		"K3S_URL":           "",
+	}
+	if masterIP != "" {
+		envVars["K3S_URL"] = fmt.Sprintf("https://%s:6443", masterIP)
+	}
+	workload := gridtypes.Workload{
+		Version: 0,
+		Name:    gridtypes.Name(k.Name),
+		Type:    zos.ZMachineType,
+		Data: gridtypes.MustMarshal(zos.ZMachine{
+			FList: k.Flist,
+			Network: zos.MachineNetwork{
+				Interfaces: []zos.MachineInterface{
+					{
+						Network: gridtypes.Name(cluster.NetworkName),
+						IP:      net.ParseIP(k.IP),
+					},
+				},
+				PublicIP:  gridtypes.Name(publicIPName),
+				Planetary: k.Planetary,
+			},
+			ComputeCapacity: zos.MachineCapacity{
+				CPU:    uint8(k.CPU),
+				Memory: gridtypes.Unit(uint(k.Memory)) * gridtypes.Megabyte,
+			},
+			Entrypoint: "/sbin/zinit init",
+			Mounts: []zos.MachineMount{
+				{Name: gridtypes.Name(diskName), Mountpoint: "/mydisk"},
+			},
+			Env: envVars,
+		}),
+	}
+	K8sWorkloads = append(K8sWorkloads, workload)
+
+	return K8sWorkloads
+}
+
+func (k *K8sCluster) ValidateToken(ctx context.Context) error {
+	if k.Token == "" {
+		return errors.New("empty token is now allowed")
+	}
+
+	is_alphanumeric := regexp.MustCompile(`^[a-zA-Z0-9]*$`).MatchString(k.Token)
```

Comment:
i think we should add a repeat for the regex group, that also matches empty string which we flagged as an error already in 186